File: BackEnd/src/domain/plan-item.ts
import {
  CurriculumItemId,
  PlanId,
  PlanItemId,
  SessionId,
  SubjectId,
} from './ids';

export enum PlanItemStatus {
  PLANNED = 'Planned',
  COMPLETED = 'Completed',
  POSTPONED = 'Postponed',
  CANCELLED = 'Cancelled',
}

export enum PlanItemPriority {
  HIGH = 'HIGH',
  MID = 'MID',
  LOW = 'LOW',
}

export class PlanItem {
  planItemId: PlanItemId;
  planId: PlanId;
  sessionId: SessionId;
  curriculumItemId: CurriculumItemId;
  subjectId: SubjectId;
  title: string;
  unitNo: number;
  lessonNo: number;
  orderInLesson: number;
  estimatedTime: number;
  originalEstimatedTime?: number;
  minEstimatedTime?: number;
  priority?: PlanItemPriority;
  isCompressible?: boolean;
  status: PlanItemStatus;
  originalSessionId?: string;
  originalSessionOrder?: number;
  carriedForwardCount?: number;
  notes?: string;
}

File: Frontend/GP/Models/PlanItem.cs
using System.ComponentModel.DataAnnotations;

namespace GP.Models
{
    public class PlanItem : BaseEntity
    {
        [Key]
        public int PlanItemId { get; set; }
        public int PlanId { get; set; }
        public Plan Plan { get; set; }
        public int SessionId { get; set; }
        public Session Session { get; set; }
        public int  CurriculumItemId { get; set; }
        public CurriculumItem CurriculumItem { get; set; }
        public int SubjectId { get; set; }
        public Subject Subject { get; set; }
        public string Title { get; set; }
        public int UnitNo { get; set; }
        public int LessonNo { get; set; }
        public int OrderInLesson { get; set; }
        public int EstimatedTime { get; set; }
        public string Status { get; set; }
        public int OriginalSessionId { get; set; }
        public OriginalSession OriginalSession { get; set; }
        public int? OriginalSessionOrder { get; set; }
        public int? CarriedForwardCount { get; set; }
        public string? Notes { get; set; }
    }
}
